src/exceptions.h

```diff
@@ -28,7 +28,12 @@ namespace HeraVM {
 
 class OutOfGasException : public std::exception {
 public:
-  const char* what() const noexcept override { return "Out of gas."; }
+  explicit OutOfGasException(std::string const& _msg = "Out of gas."):
+    msg(std::move(_msg))
+  {}
+  const char* what() const noexcept override { return msg.c_str(); }
+private:
+  std::string msg;
```

Comment:
If you don't plan to have runtime created messages, you can just keep the reference to a string literal.

Reply:
See the next PR depending on this for the reason.